jdk25u-jdk-25.0.3-ga/src/hotspot/os/solaris/osThread_solaris.hpp:35:11: error: 'thread_t' does not name a type; did you mean 'pthread_t'?
   35 |   typedef thread_t thread_id_t;
      |           ^~~~~~~~
      |           pthread_t

--- jdk25u-jdk-25.0.3-ga/src/hotspot/os/solaris/osThread_solaris.hpp.old
+++ jdk25u-jdk-25.0.3-ga/src/hotspot/os/solaris/osThread_solaris.hpp
@@ -28,6 +28,7 @@
 #include "runtime/osThreadBase.hpp"
 #include "suspendResume_posix.hpp"
 #include "utilities/globalDefinitions.hpp"
+#include <thread.h>
 
 class OSThread : public OSThreadBase {
   friend class VMStructs;
